tests/lib/nested.sh

```diff
@@ -574,7 +574,7 @@ nested_create_core_vm() {
                     fi
 
                 elif nested_is_core_20_system; then
-                    snap download --basename=pc-kernel --channel="20/edge" pc-kernel
+                    snap download --basename=pc-kernel --channel="20/$NESTED_KERNEL_CHANNEL" pc-kernel
```

Comment:
I think this change is actually correct now, as you found actually the other PR does not need env vars at all, and it may be generally useful to be able to specify the channels that the repacked snaps are coming from, so while this change isn't correct for my test, I think this change is correct/useful in general